src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -35,10 +36,51 @@
      */
     boolean authenticate(InetAddress remoteAddress, int remotePort);
 
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(Certificate[] certificates)
+    {
+        return false;
+    }
```

Comment:
I don't think you need this new method. It would be up to the implementor to write an `authenticator` method that checks the certificates if that's what they need.